MoreLinq.Test/TransposeTest.cs

```diff
@@ -0,0 +1,225 @@
+                new [] { 10, 20, 30 },
+                new [] { 11, 21, 31 },
+                new [] { 12, 22, 32 },
+                new [] { 13, 23, 33 },
+            };
+
+            using (var seq1 = TestingSequence.Of(10, 11, 12, 13))
+            using (var seq2 = TestingSequence.Of(20, 21, 22, 23))
+            using (var seq3 = TestingSequence.Of(30, 31, 32, 33))
+            using (var matrix = TestingSequence.Of(seq1, seq2, seq3))
+            {
+                AssertMatrix(matrix.Transpose(), expectations);
+            }
+        }
+
+        [Test]
+        public void TransposeWithDifferentsLengthEnumerables()
+        {
+            var expectations = new[]
+            {
+                new int[] { 10, 20, 30 },
+                new int[] { 11, 31 },
+                new int[] { 32 }
+            };
+
+            using (var seq1 = TestingSequence.Of(10, 11))
+            using (var seq2 = TestingSequence.Of(20))
+            using (var seq3 = TestingSequence.Of<int>())
+            using (var seq4 = TestingSequence.Of(30, 31, 32))
+            using (var matrix = TestingSequence.Of(seq1, seq2, seq3, seq4))
+            {
+                AssertMatrix(matrix.Transpose(), expectations);
+            }
+        }
+
+        [Test]
+        public void TransposeAssertPositions()
+        {
+            var matrix = new[]
+            {
+                new int[] { 10, 11 },
+                new int[] { 20 },
+                new int[] { },
+                new int[] { 30, 31, 32 }
+            };
+
+            var traspose = matrix.Transpose();
+
+            Assert.That(matrix.Last().Last(), Is.EqualTo(traspose.Last().Last()));
+            Assert.That(matrix.First().First(), Is.EqualTo(traspose.First().First()));
+        }
+
+        [Test]
+        public void TransposeWithAllInnerSequencesInfinite()
+        {
+            var matrix = MoreEnumerable.Generate(1, x => x + 1)
+                                       .Where(x => isPrime(x))
+                                       .Take(3)
+                                       .Select(x => MoreEnumerable.Generate(x, n => n * x));
+
+            var result = matrix.Transpose().Take(5);
+
+            var expectations = new[]
+            {
+                new[] { 2,    3,    5 },
+                new[] { 4,    9,   25 },
+                new[] { 8,   27,  125 },
+                new[] { 16,  81,  625 },
+                new[] { 32, 243, 3125 }
+            };
+
+            AssertMatrix(result, expectations);
+        }
+
+        public void TransposeWithSomeSequencesInfinite()
+        {
+            var matrix = MoreEnumerable.Generate(1, x => x + 1)
+                                       .Where(x => isPrime(x))
+                                       .Take(3)
+                                       .Select(x => x == 3 ? MoreEnumerable.Generate(x, n => n * x)
+                                                           : MoreEnumerable.Generate(x, n => n * x).Take(3));
```

Comment:
Doesn't the `Take(3)` here belong in the true part of the conditional expression?